compiler/rustc_typeck/src/check/method/suggest.rs

```diff
@@ -1916,8 +1916,8 @@ impl Ord for TraitInfo {
         // Local crates are more important than remote ones (local:
         // `cnum == 0`), and otherwise we throw in the defid for totality.
 
-        let lhs = (other.def_id.krate, other.def_id);
-        let rhs = (self.def_id.krate, self.def_id);
+        let (lhs, _) = (other.def_id.krate, other.def_id);
+        let (rhs, _) = (self.def_id.krate, self.def_id);
         lhs.cmp(&rhs)
```

Comment:
This impl is wrong and will probably produce surprising results.
We implement `Eq`, so we expect a real equality, not just `CrateNum`.
Is the `PartialOrd`/`Ord` bound used? Where? Can we live without?

Reply:
The ordering is used here https://github.com/rust-lang/rust/blob/088dd7cb54f57bd1c870a343cb5dc3d3d2703761/compiler/rustc_typeck/src/check/method/suggest.rs#L1651 and in the `PartialEq` implementation: https://github.com/rust-lang/rust/blob/088dd7cb54f57bd1c870a343cb5dc3d3d2703761/compiler/rustc_typeck/src/check/method/suggest.rs#L1906

Reply:
And the `Eq`,`PartialEq` are currently used for deduplication: https://github.com/rust-lang/rust/blob/088dd7cb54f57bd1c870a343cb5dc3d3d2703761/compiler/rustc_typeck/src/check/method/suggest.rs#L1652

Reply:
Would it work or defeat our purpose to sort the TraitInfos by DefIndex?